src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs

```diff
@@ -91,8 +91,23 @@ internal static void ApplyValidationAttributes(this JsonNode schema, IEnumerable
             }
             else if (attribute is RangeAttribute rangeAttribute)
             {
-                schema[OpenApiSchemaKeywords.MinimumKeyword] = decimal.Parse(rangeAttribute.Minimum.ToString()!, CultureInfo.InvariantCulture);
-                schema[OpenApiSchemaKeywords.MaximumKeyword] = decimal.Parse(rangeAttribute.Maximum.ToString()!, CultureInfo.InvariantCulture);
+                // Use InvariantCulture if explicitly requested or if the range has been set via the
+                // RangeAttribute(double, double) or RangeAttribute(int, int) constructors.
+                var targetCulture = rangeAttribute.ParseLimitsInInvariantCulture || rangeAttribute.Minimum is double || rangeAttribute.Maximum is int
+                    ? CultureInfo.InvariantCulture
+                    : CultureInfo.CurrentCulture;
+
+                var minString = rangeAttribute.Minimum.ToString();
+                var maxString = rangeAttribute.Maximum.ToString();
+
+                if (decimal.TryParse(minString, NumberStyles.Any, targetCulture, out var minDecimal))
```

Comment:
Is this user-friendly to not set any minimum\maximum value, and not throw some specific exception/set some default value/maybe even write an analyzer to let users know this will not succeed?

_asking because I dont know the possible mechanisms of interaction with users in case of OpenAPI_

Reply:
> Is this user-friendly to not set any minimum\maximum value, and not throw some specific exception

I prefer not setting anything to throwing an exception since there's really no way for users to catch the exception and modify how the document gets created. By not setting a value when we can't parse, we give users the option of setting it via the transformer APIs (see [here](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/openapi/customize-openapi)).

> set some default value

It's hard to know what a good default value should be so setting it to `null` feels appropriate.

> maybe even write an analyzer to let users know this will not succeed?

This specifically won't succeed when used in OpenAPI scenarios so our analyzer would have to be selective about when it warns and the message/verbosity it provides.

Reply:
I'm with @captainsafia here. The current implementation throwing an exception is one of the blockers for us to move from Swashbuckle to the new ASP.NET OpenAPI implementation.

I'm not sure if it's feasible, but logging a message could help diagnose _why_ the minimum/maximum isn't set. But otherwise, I'd rather have it not set than have an exception that I can't fix in any way (except for removing the attribute which also removes the validation!).

Reply:
makes sense, thanks for explanation!